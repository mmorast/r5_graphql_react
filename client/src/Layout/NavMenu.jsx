import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'


const links = [
  {to: "/", text: "Home"},
  {to: "/posts", text: "Posts"},
]

const NavMenu = props => {

  return (
    <NavDiv>
      {links.map((link, idx) => {
        return <StyledLink key={`link-${idx}`} to={link.to}>{link.text}</StyledLink>
      })}
    </NavDiv>
  )
}

export default NavMenu

const NavDiv = styled.nav`
  padding: 10px;
`

const StyledLink = styled(Link)`
  padding: 5px;
  color: white;
`
