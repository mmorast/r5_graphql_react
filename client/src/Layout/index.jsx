import React from 'react'
import styled from 'styled-components'
import NavMenu from './NavMenu'

const Layout = props => {
  return (
    <LayoutWrapper>
      <FullHeight>
        <AppHeader>
          <NavMenu/>
        </AppHeader>
        <AppBody>
          {props.children}
        </AppBody>
      </FullHeight>
    </LayoutWrapper>
  )
}

export default Layout

const LayoutWrapper = styled.div`
  height: 100%;
`

const FullHeight = styled.div`
  min-height: 100%;
  display: flex;
  flex-direction: column;
`

const AppHeader = styled.header`
  flex: 1;
`

const AppBody = styled.main`
  flex: 7;
  margin: 50px auto;
  width: 70%;
`

