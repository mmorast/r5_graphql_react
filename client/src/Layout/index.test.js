import React from 'react'
import {shallow} from 'enzyme'
import Layout from './index'

test("Renders Layout without blowing up", () => {
  let actual = shallow(<Layout/>)
  expect(actual).toMatchSnapshot()
})
