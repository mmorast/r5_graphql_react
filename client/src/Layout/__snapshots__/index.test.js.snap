// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Layout without blowing up 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        <styled.header>
            <NavMenu />
        </styled.header>
        <styled.main />
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            <styled.header>
                  <NavMenu />
            </styled.header>
            <styled.main />
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout />,
    "_instance": null,
    "_newState": null,
    "_rendered": <styled.div>
      <styled.div>
            <styled.header>
                  <NavMenu />
            </styled.header>
            <styled.main />
      </styled.div>
</styled.div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout />,
}
`;
